File: engine/parseUtils.js
var readline = require ('readline');
var fs = require ('fs');

const SECONDS_IN_HOURS = 3600;
const SECONDS_IN_MINUTES = 60;

/**
 * Parses the timetable output for the video
 */
exports.parseTimetable = function (timetablePath, cb) {

    console.log (timetablePath);
    let timeArray = [];

    const rl = readline.createInterface({
        input: fs.createReadStream(timetablePath)
    });

    rl.on ('line', function (line) {
        let tokens = line.split (':');

        let hours = parseInt(tokens[1]);
        let minutes = parseInt(tokens[2]);
        let seconds = parseInt(tokens[3]);

        let timeInSeconds = hours * SECONDS_IN_HOURS +
                            minutes * SECONDS_IN_MINUTES +
                            seconds;

        timeArray.push (timeInSeconds);

    })

    rl.on ('close', function () {
        cb(timeArray);
    });
}


/**
 * Parses the text contents for the video
 */
exports.parseContents = function (contentsDir, length) {

    let contentsArray = [];

    for (var i = 1; i <= length; i++) {
        let filename = contentsDir + i + '.txt';
        var data = fs.readFileSync (filename, 'utf-8');

        // replace newlines with space
        data = data.replace(/\n/g, " ");

        contentsArray.push (data.trim());
    }

    //console.log (contentsArray);
}

File: engine/parseUtils.py
SECONDS_IN_HOURS = 3600;
SECONDS_IN_MINUTES = 60;

def parseTimetable (timetablePath): 

    timeArray = []

    with open(timetablePath) as timetable:
            for line in timetable:
                tokens = line.split(':')

                hours = int(tokens[1])
                minutes = int(tokens[2])
                seconds = int(tokens[3].split('.')[0])

                timeInSeconds = hours * SECONDS_IN_HOURS + \
                                minutes * SECONDS_IN_MINUTES + \
                                seconds

                timeArray.append (timeInSeconds)

    return timeArray

def parseContents (contentsDir, length):

    contentsArray = []
    
    for i in range(1, length+1):
        filename = contentsDir + str(i) + '.txt'

        with open(filename, 'r') as contentsFile:
            data=contentsFile.read().replace('\n', '')
            contentsArray.append (data.strip())

    return contentsArray
